plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -86,7 +86,7 @@ private Module getGlueStatisticsModule(Class<? extends GlueColumnStatisticsProvi
     @ForGlueHiveMetastore
     public Executor createExecutor(CatalogName catalogName, GlueHiveMetastoreConfig hiveConfig)
     {
-        return createExecutor("hive-glue-partitions-%s", hiveConfig.getWriteStatisticsThreads());
+        return createExecutor("hive-glue-partitions-%s", hiveConfig.getGetPartitionThreads());
```

Comment:
This will create with correct configs but this executor is not used in the `getPartitions` call.

You will also need to change `getPartitions` in `GlueHiveMetastore` to use the non-stats executor. i.e. the line https://github.com/trinodb/trino/pull/6178/files#diff-d827fd361f2fef94d35ef8593d0f9a7f168366f28acdb88a67ff413b1904a3d0R784 needs to be reverted.

Reply:
No. This is not true. 
Executor passed to `GlueHiveMetastore` is `@ForGlueHiveMetastore Executor partitionsReadExecutor`.
Which is exactly what we are binding here. Please drop the other commit which does just renames (IMO current naming is more accurate)

Reply:
@losipiuk thanks for point that. all done